common.js

```diff
@@ -170,8 +171,19 @@ export function setupPath(newPathEntries) {
   // Then add new path entries using core.addPath()
   let newPath
   if (windows) {
+    // main Ruby dll determines whether mingw or ucrt build
+    let build_sys = 'mingw64'
+    const cmd = `dir ${newPathEntries[0]}\\x64-ucrt-ruby*.dll /B`
```

Comment:
Could you find a way to do this without a subprocess, e.g., using some globbing from JavaScript?

Reply:
I looked for a js 'glob' method/function, couldn't find one.  If you switch on time stamps, this part of the code didn't even register one second in time?

Reply:
I think we should simply list files under that directory and check if one matches then, in a separate function.
A subcommand for this relying on the shell and shell globbing feels very fragile and would break e.g. if there was any unintended space in there.

Reply:
> if there was any unintended space in there.

Good point, but how?  Regardless, I'll add quotes.  I don't understand your aversion to using a 'subcommand'.  It's a standard shell command, it doesn't take any time, and it's been around since DOS days.  A lot simpler that a separate function.

Detecting whether a Windows Ruby is mingw or ucrt may change...

Reply:
> I don't understand your aversion to using a 'subcommand'.

This is my requirement, avoiding the subshell is simply more reliable.
A subprocess for this is total overkill, makes it significantly less reliable (e.g., what if there is some extr aouput during shell startup, how we do know which shell is started, if quoting remains correct, etc).
I also find it pretty hard to read.

Reply:
Also this whole logic feels quite messy in the middle of `setupPath`, so I'd want a separate function anyway, and at point might as well do it cleanly without a subshell.

Reply:
> This is my requirement

I'll see what can be done using the existing node packages.

> A subproccess for this is total overkill, makes it significantly less reliable

7 to 25 mS, and a command that's existed since DOS days.  The command used (below) is the cmd equivalent of `Dir.glob`, which isn't available in standard node.js.

```cmd
dir ${newPathEntries[0]}\\x64-ucrt-ruby*.dll /B
```

I could add a regex match on the return/output.  Is that ok, or you still demand losing the `execSync` call?

**Off-topic** - I wrote code on Apple II's and the original IBM & Compaq pc's...